blog/_posts/2021-07-09-FastAI-tabular-data.md

```diff
@@ -0,0 +1,207 @@
+If your data is present in a different format, you could use a package which supports loading that format, provided that the final object created supports the required interface.
+
+```julia
+julia> using CSV, DataFrames
+
+julia> using FastAI, FastAI.Datasets
+
+julia> path = joinpath(datasetpath("adult_sample") , "adult.csv");
+
+julia> df = CSV.File(path)|> DataFrames.DataFrame; first(df, 5)
+5×15 DataFrame
+ Row │ age    workclass          fnlwgt  education     education-num  marital-status       occupation        rel ⋯
+     │ Int64  String             Int64   String        Float64?       String               String?           Str ⋯
+─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────
+   1 │    49   Private           101320   Assoc-acdm            12.0   Married-civ-spouse  missing            Wi ⋯
+   2 │    44   Private           236746   Masters               14.0   Divorced             Exec-managerial   No
+   3 │    38   Private            96185   HS-grad          missing     Divorced            missing            Un
+   4 │    38   Self-emp-inc      112847   Prof-school           15.0   Married-civ-spouse   Prof-specialty    Hu
+   5 │    42   Self-emp-not-inc   82297   7th-8th          missing     Married-civ-spouse   Other-service     Wi ⋯
+                                                                                                 8 columns omitted
+```
+
+After getting an object satisfying Tables.jl Interface, we can pass this to `FastAI.Datasets.TableDataset` to get a container satisfying the `LearnBase` Interface.
+
+```julia
+julia> td = TableDataset(df)
+TableDataset{DataFrame}(32561×15 DataFrame
+   Row │ age    workclass          fnlwgt  education      education-num  marital-status       occupation         ⋯
+       │ Int64  String             Int64   String         Float64?       String               String?            ⋯
+───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────
+     1 │    49   Private           101320   Assoc-acdm             12.0   Married-civ-spouse  missing            ⋯
+     2 │    44   Private           236746   Masters                14.0   Divorced             Exec-managerial
+   ⋮   │   ⋮            ⋮            ⋮           ⋮              ⋮                 ⋮                   ⋮          ⋱
+ 32560 │    32   Local-gov         217296   HS-grad                 9.0   Married-civ-spouse   Transport-moving
+ 32561 │    26   Private           182308   Some-college           10.0   Married-civ-spouse   Prof-specialty
+                                                                                  8 columns and 32557 rows omitted)
+
+```
+
+What this `TableDataset` object allows us to do is that we can get any observation at a particular index by using `getindex(td, index)` and the total number of observations by using `nobs(td)`.
```

Comment:
Maybe add in a line about why this is cool, and how it generalises the usual `getindex` based approach for arrays to data frames?